src/LocalBackend/LocalLibrary.vala

```diff
@@ -565,10 +565,20 @@ public class Noise.LocalLibrary : Library {
                 sql.select_add_field ("rowid", null, null);
                 Gda.SqlBuilderId[] ids = null;
 
-                string[] fields = {"title", "artist", "composer", "album_artist", "album", "grouping", "comment"};
+                /* Enabling directory/path/uri search through SearchEntry */
```

Comment:
Can you clarify what the role of this code is and why uri can't be included with the other fields? Can you have a search by uri and by title at the same time?

Reply:
Think of a situation where you have a 
1. music file with title "English Song" (/home/someone/musics/latest songs/1.mp3) and 
2. a directory "/home/someone/musics/English Songs", 
okay. 

Now when you type "uri:english songs" in searchField then depending on "string[] fields" the result varies.

For example, if "uri" field included with other fields then the result will contain both the paths and if we use only "uri" field (excluding other fields) then our result will only contain that directory path. (which is goal of uri: prefix search).

Let me know if you need to change it to mix both results but that will not help users to create separate result for particular directory.

Thanks :D

Reply:
Primary goal for "uri:" filter is to "list the music from particular directory/path" and "create smartplaylist from uri filter".

:D

Reply:
Sure but there are implementation limitations, not conceptual limitations. For all I can tell, there's nothing limiting an implementation where a query like `uri:"/some/path/here" artist:"Andrei Zisu"` should not be possible.

Reply:
I didn't understand. Can you please clarify what are you saying ?

Am I supposed to remove this filter or what !?

Thanks :D

Reply:
Ok so it's my understanding that you can't use `uri` and `artist` at the same time, right? Why is that?

Reply:
No, I/We can use `uri`, `artist` or `any-field` at same time, It will be simple, but it will mix-up the result. 

Actually I got this idea when I started using noise and then I was searching for the option to filter search result on location only (as most of the time I usually play music directory wise) but didn't found one. So implemented it.

We can also combine different filters in a single query just like you suggested `uri:"/path/to/musics" artist:"Andrei Zisu"`.

The only reason to not include `uri` with other fields is to separate music from that particular path. 

:D

Reply:
Sorry for the long delay - was a bit busy.

Well if you wanna filter by location only - then you just use the URI filter and nothing else.

Reply:
Honestly I don't see any reason why this filter should be singled out from the rest.

Reply:
No worries about delay :)

Okay, I will mix up the result but then in some situation users have to type full path and also we don't need `uri:` prefix now.

I think you're right in a sense because without using `uri:`, it would be simple for end users also.

Reply:
> but then in some situation users have to type full path and also we don't need uri: prefix now.

What do you mean?

Reply:
Cool stuff. I'll have a final look and then approve this pull request.

Reply:
Okay :)

Now my mid-term exams are gonna start, bye. ;)